core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -3199,7 +3192,23 @@ private void transitiveInference(RelOptRule... extraRules) throws Exception {
     checkPlanning(tester, preProgram, new HepPlanner(program), sql);
   }
 
-  @Test public void testSwapOuterJoin() {
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-2249">[CALCITE-2249]
+   * AggregateJoinTransposeRule generates inequivalent nodes if Aggregate relNode contains
+   * distinct aggregate function.</a>. */
+  @Test public void testPushDistinctAggregateIntoJoin() throws Exception {
+    final HepProgram preProgram = new HepProgramBuilder()
+            .addRuleInstance(AggregateProjectMergeRule.INSTANCE)
+            .build();
```

Comment:
`preProgram` is not used, can we remove it?

Reply:
OK, thanks.